provisioner/aws.go

```diff
@@ -240,7 +240,7 @@ func (a *awsAdapter) CreateOrUpdateClusterStack(parentCtx context.Context, stack
 
 		// TODO(tech-depth): this is only used for migrating from
 		// legacy node pools to real node pool support.
-		if nodePoolFeatureEnabled(cluster) {
+		if asg != nil && nodePoolFeatureEnabled(cluster) {
```

Comment:
When the legacy pools get removed with the next PR ASG will be nil and we shouldn't fail.